cpp/src/arrow/array.h

```diff
@@ -1165,12 +1166,21 @@ class ARROW_EXPORT UnionArray : public Array {
   /// Note that this buffer does not account for any slice offset
   std::shared_ptr<Buffer> type_ids() const { return data_->buffers[1]; }
 
+  const type_id_t* raw_type_ids() const { return raw_type_ids_ + data_->offset; }
```

Comment:
@wesm I was wondering... should we rename these methods to `type_codes` and `raw_type_codes`? There is a bit of a mixed terminology here (`UnionType` uses `type_codes`, also `type_id` is used for something different).

Reply:
I'd be in favor of changing the names to be more clear and where relevant, conforming to the specification

Reply:
Well, the specification uses "types" and "type ids" :-/